### PR TITLE
Added support for downloading sdks from multiple archives repositories

### PR DESCRIPTION
**Pull request by JCash**

### CODE REVIEW

File: server/src/main/java/com/defold/extender/services/DefoldSdkService.java [hunk 1]
```diff
@@ -102,26 +102,38 @@ public File getRemoteSdk(String hash) throws IOException, URISyntaxException, Ex
...
             if (lockFile.createNewFile()) { // atomic creation of lock file
                 try {
-                    LOGGER.info("Downloading Defold SDK version {} ...", hash);
+                    boolean sdkFound = false;
+                    for (String url_pattern : REMOTE_SDK_URL_PATTERNS) {
 
-                    URL url = new URL(String.format(REMOTE_SDK_URL_PATTERN, hash));
-                    ClientHttpRequestFactory clientHttpRequestFactory = new SimpleClientHttpRequestFactory();
-                    ClientHttpRequest request = clientHttpRequestFactory.createRequest(url.toURI(), HttpMethod.GET);
+                        URL url = new URL(String.format(url_pattern, hash));
```
Comment (JCash) [inline server/src/main/java/com/defold/extender/services/DefoldSdkService.java:108]: Loop over the patterns, and try one url after another.

**Comment by JCash** [comment]
Not sure, I'll look into it.
Do you have such a test url?

**Comment by britzl** [comment]
Yes, d.defold.com/archive/6cbddff93b77472c076dce83217e5d2b824236b1/engine/x86_64-linux/dmengine_release

**Comment by JCash** [comment]
Testing with that version (6cbddff93b77472c076dce83217e5d2b824236b1) worked with the current code.

**Pull request merged**